### PR TITLE
libjpeg-turbo: Fix Darwin lib install name

### PR DESCRIPTION
**Pull request by climbfuji**
## Description

The libjpeg-turbo library has a bad install name on macOS:
```
> otool -L envs/unified-env/install/apple-clang/13.1.6/libjpeg-turbo-2.1.0-3wyhtqp/lib/libjpeg.62.3.0.dylib 
envs/unified-env/install/apple-clang/13.1.6/libjpeg-turbo-2.1.0-3wyhtqp/lib/libjpeg.62.3.0.dylib:
	@rpath/libjpeg.62.dylib (compatibility version 62.0.0, current version 62.3.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1311.100.3)
```
This PR fixes it:
```
> otool -L envs/unified-env/install/apple-clang/13.1.6/libjpeg-turbo-2.1.0-3wyhtqp/lib/libjpeg.62.3.0.dylib 
envs/unified-env/install/apple-clang/13.1.6/libjpeg-turbo-2.1.0-3wyhtqp/lib/libjpeg.62.3.0.dylib:
	/Users/heinzell/work/spack-stack/spack-stack-met_graphics/envs/unified-env/install/apple-clang/13.1.6/libjpeg-turbo-2.1.0-3wyhtqp/lib/libjpeg.62.3.0.dylib (compatibility version 62.0.0, current version 62.3.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1311.100.3)
```

### CODE REVIEW

**Comment by climbfuji** [comment]
Sorry to ask for your review here @tldahlgren, please let me know if I should add someone else instead (in the absence of a package maintainer). Thanks!

File: var/spack/repos/builtin/packages/libjpeg-turbo/package.py
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/libjpeg-turbo/package.py]: If this really applies to the shared library, you may want an extra check:

```suggestion
        if self.spec.satisfies("platform=darwin") and ("+shared" in self.spec):
```

**Review by tldahlgren — COMMENTED** [review]
I think you may want to make sure you're building the shared library.

File: var/spack/repos/builtin/packages/libjpeg-turbo/package.py
Comment (climbfuji) [reply var/spack/repos/builtin/packages/libjpeg-turbo/package.py]: Good point, thanks!

**Review by tldahlgren — APPROVED** [review]
LGTM

**Comment by tldahlgren** [comment]
Updating branch since the CI issue holding this PR up has been fixed.

**Pull request merged**
